### ISSUE TITLE
Missing potential_vorticity and relative_vorticity: expose functional cgrid operators for shallow water and QG models

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/finitevolX#144]
## What's missing?

somax exposes functional `potential_vorticity` and `relative_vorticity` operators for C-grid shallow water and quasi-geostrophic models, supporting both shallow water (SWM) and QG variants. finitevolX does not expose these as public functional API matching somax, which blocks direct refactor of potential vorticity-related logic. These operators should be available as top-level functions.

### Required public exports:
- `relative_vorticity(u, v, dx, dy): Float[Array, "Nx-1 Ny-1"]`
  - SWM: dv/dx - du/dy (face difference to node)
- `potential_vorticity(h, u, v, dx, dy, params, domain, mask=None): Float[Array, "Nx+1 Ny+1"]`
  - SWM: (ζ + f) / h, where ζ = relative_vorticity
  - Support both planetary and relative vorticity
  - Accept grid domain, physical params, mask for ocean/land
- `potential_vorticity_qg(psi, step_size, alpha=1.0, beta=0.0, f=None): Float[Array, "Nx-1 Ny-1"]`
  - QG: α∇²ψ + βψ + f, batched for multilayer if needed

## Reference somax implementation:
- [somax/_src/models/sw/operators.py](https://github.com/jejjohnson/somax/blob/8dc7229fa57dde0366986b97b6fe7372e3da21e0/somax/_src/models/sw/operators.py):
  ```python
  def relative_vorticity(u, v, dx, dy):
      dv_dx = difference(v, axis=0, step_size=dx)
      du_dy = difference(u, axis=1, step_size=dy)
      ζ = dv_dx - du_dy
      return ζ
  def potential_vorticity(h, u, v, dx, dy, params, domain, mask):
      f_on_q = params.coriolis_f0 + domain.grid_axis[1] * params.coriolis_beta
      vort_r = relative_vorticity(u, v, dx, dy)
      h_on_q = center_avg_2D(h)
      q = (vort_r + f_on_q) / h_on_q
      if mask is not None:
          q *= mask.values
      return q
  ```
- [somax/_src/models/qg/operators.py](https://github.com/jejjohnson/somax/blob/8dc7229fa57dde0366986b97b6fe7372e3da21e0/somax/_src/models/qg/operators.py)
  ```python
  def potential_vorticity(psi, step_size, alpha=1.0, beta=0.0, f=None):
      q = alpha * laplacian(psi, step_size=step_size)
      if pad_bc_fn is not None:
          q = pad_bc_fn(q)
      else:
          q = jnp.pad(q, pad_width=((1,1),(1,1)))
      if beta != 0.0:
          q += beta * psi
      if f is not None:
          q += f
      q = center_avg_2D(q)
      return q
  ```

## Tasks
- [ ] Implement `relative_vorticity` operator for C-grid SWM
- [ ] Implement `potential_vorticity` operator for C-grid SWM with planetary vorticity, masking
- [ ] Implement `potential_vorticity_qg` operator (batched, multilayer QG)
- [ ] Add public API & documentation
- [ ] Add tests verifying numerical results against somax

---
### Priority
High — needed for vorticity, PV refactor and cross-repo compatibility
### Labels
- enhancement
